plugins/puppetdb/puppetdb

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/python3
+#
+# This code is licensed under GPLv3+
+#
+# Create a graph out of PuppetDB's JVM memory usage.
+#
+# dependencies: python3-requests
+#
+# compatibility:
+#  * PuppetDB 6.x: https://puppet.com/docs/puppetdb/6.0/api/metrics/v1/mbeans.html#jvm-metrics
+#
+# Configuration can be done via environment variables, usually via a munin
+# configuration file for this plugin (e.g.
+# /etc/munin/plugin-conf.d/puppetdb.conf)
+#
+#  * pdburl -- Set the URL to your PuppetDB instance. This url should point to
+#              the mbeans endpoint. By default this has a value of
+#              http://localhost:8080/metrics/v1/mbeans
+#
+#  * timeout -- Time in seconds (int) to wait for a result when querying the REST
+#               API. By default, wait for 2 seconds
+#
+import os
+import sys
+import requests
```

Comment:
Please consider using python's `urllib` and `json` module instead, since they are available in every python installation.
Your GET request looks simple enough for an easy switch.
(personally I consider `requests` to be useful only for interactive usage - see [requests#4842](https://github.com/psf/requests/issues/4842) for an example of non-deterministic behavior depending on locally installed modules)

Reply:
`urllib` on the other hand does not become trivial, as soon as you want to add certificate verification and client cert & key, which is what you need once the puppetdb is only available through SSL. Using `requests, this stays really simple.

Reply:
Indeed the certificate validation was problematic in older versions. But since Python 3.4 (see [release notes](https://docs.python.org/3/whatsnew/3.4.html#pep-476-enabling-certificate-verification-by-default-for-stdlib-http-clients)) certificates are verified by default (no additional arguments are required). Since you are using Python's new format strings, you may assume, that a Python version with proper certificate handling is in use.
The users of this munin plugin would surely appreciate, that the plugin works out of the box (without installing an additional python package).
Anyway: feel free to decide, whether you want to stick to `requests` or switch to `urllib`.

btw.: do you think, that it is reasonable to assume that the puppeddb API is configured with a valid certificate (in case of HTTPS)? Otherwise you may want to introduce a configuration setting for disabling certificate checks.

Reply:
I did not implement it because my use case is checking a local puppetdb system, bu I do think that verifying certificates is an important feature for this plugin since many folks have their puppetdb systems on a different server than their puppetmaster

Reply:
OK. So we assume, that people using https for connecting to the puppetdb use a proper certificate. Good.